Libraries/PeriphDrivers/Source/SPIMSS/spimss_reva.c

```diff
@@ -231,6 +242,197 @@ int MXC_SPIMSS_RevA_MasterTrans(mxc_spimss_reva_regs_t *spi, spimss_reva_req_t *
+
+            if ((true == states[i].auto_dma_handlers) && (0 == (MXC_DMA)->ch[ch].cnt)) {
+                //Disable interrupts.
+                MXC_DMA_DisableInt(ch);
+                MXC_DMA_SetChannelInterruptEn(ch, false, false); // Disable ctz interrupt.
+                MXC_DMA_ReleaseChannel(ch);
+                states[i].spi->dma &= ~(1 << 15); // Disable SPIMSS TX DMA requests
+                states[i].channelTx = -1;
+            }
+
+            if (NULL != temp_req->callback) {
+                temp_req->callback(temp_req, E_NO_ERROR);
+            }
+            break;
+        } else if (ch == states[i].channelRx) {
+            states[i].req->rx_num = transaction_size - (MXC_DMA)->ch[ch].cnt;
+
+            if ((true == states[i].auto_dma_handlers) && (0 == (MXC_DMA)->ch[ch].cnt)) {
+                //Disable interrupts.
+                MXC_DMA_DisableInt(ch);
+                MXC_DMA_SetChannelInterruptEn(ch, false, false); // Disable ctz interrupt.
+                MXC_DMA_ReleaseChannel(ch);
+
+                if (temp_req->deass) {
+                    states[i].spi->mode |= ((uint32_t)(0x1UL << 0)); // Set Slave Select to HIGH.
+                }
+                states[i].spi->ctrl &= ~(uint32_t)(0x1UL << 0); // Disable SPIMSS transaction.
+
+                states[i].spi->dma &= ~(1 << 31); // Disable SPIMSS RX DMA requests
+
+                states[i].channelRx = -1;
+                states[i].auto_dma_handlers = false;
+                states[i].req = NULL;
+            }
+
+            if (NULL != temp_req->callback) {
+                temp_req->callback(temp_req, E_NO_ERROR);
+            }
+            break;
+        }
+    }
+}
+
+static mxc_dma_config_t dma_config_tx;
+static mxc_dma_config_t dma_config_rx;
+static mxc_dma_adv_config_t dma_adv_config_tx;
+static mxc_dma_adv_config_t dma_adv_config_rx;
+static mxc_dma_srcdst_t srcdst_config_tx;
+static mxc_dma_srcdst_t srcdst_config_rx;
+/* ************************************************************************** */
+int MXC_SPIMSS_RevA_MasterTransDMA(mxc_spimss_reva_regs_t *spi, spimss_reva_req_t *req)
+{
+    // Calculating the transaction size in byte.
+    int sent_byte_len = ((req->len * (req->bits / 8)) - req->tx_num);
+    int tx_channel_id = -1;
+    int rx_channel_id = -1;
+    int spi_num = MXC_SPIMSS_GET_IDX((mxc_spimss_regs_t *)spi);
+
+    //Reset static variables.
+    // tx_channel_reg = NULL;
+    // rx_channel_reg = NULL;
+    memset(&dma_config_tx, 0, sizeof(dma_config_tx));
+    memset(&dma_config_rx, 0, sizeof(dma_config_rx));
+    memset(&srcdst_config_tx, 0, sizeof(srcdst_config_tx));
+    memset(&srcdst_config_rx, 0, sizeof(srcdst_config_rx));
+    memset(&dma_adv_config_tx, 0, sizeof(dma_adv_config_tx));
+    memset(&dma_adv_config_rx, 0, sizeof(dma_adv_config_rx));
+
+    // We use SPIMSS master trans function to send data from
+    if (E_NO_ERROR != MXC_SPIMSS_RevA_TransSetup(spi, req, 1)) {
+        return -1;
```

Comment:
Instead of returning -1, return the error code that `MXC_SPIMSS_RevA_TransSetup` gives you here

Reply:
Updated.